### PR TITLE
junipersrx-module initial release

### PR DESCRIPTION
**Pull request by StefanSa**
## What does this PR do?
This PR Introduces the JuniperSRX filebeat module. Focusing currently on JuniperSRX Firewall, but should include other Juniper products as separate PR's later on.

## Why is it important?
Adding more supported products to the filebeat portfolio.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
INTEGRATION_TESTS=1 BEAT_STRICT_PERMS=false TESTING_FILEBEAT_MODULES=junipersrx nosetests -v -s tests/system/test_xpack_modules.py

### CODE REVIEW

**Comment by P1llus** [comment]
This is an initial community review that Stefan has shared that I will be helping out with

**Review by P1llus — APPROVED** [review]

**Comment by P1llus** [comment]
Jenkins test this please

**Comment by P1llus** [comment]
Had to temporarily approve it to kick off some of the CI jobs that didn't want to start without it, please ignore the approval for now.

**Comment by P1llus** [comment]
Jenkins test this please

**Comment by adriansr** [comment]
I've removed some fields under juniper.srx that were unused because the pipeline will rename them into ECS fields. Now it's below the 1MB index-pattern limit (just a few bytes left though)

File: x-pack/filebeat/module/juniper/srx/test/secintel.log-expected.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        "log.level": "informational",
+        "log.offset": 0,
+        "network.iana_number": "1",
+        "observer.egress.zone": "DMZ",
+        "observer.ingress.zone": "Outside",
+        "observer.name": "SRX-1500",
+        "observer.product": "SRX",
+        "observer.type": "firewall",
+        "observer.vendor": "Juniper",
+        "server.ip": "10.10.0.10",
```
Comment (marc-gr) [inline x-pack/filebeat/module/juniper/srx/test/secintel.log-expected.json:47]: We could populate `related.ip` and `related.hosts` for most of these I think

Comment (P1llus) [reply x-pack/filebeat/module/juniper/srx/test/secintel.log-expected.json:47]: Those should be there but seem to be missing for one or two of the ingest pipelines, will review it now and push the relevant pipeline change

**Review by marc-gr — APPROVED** [review]

**Comment by P1llus** [comment]
Jenkins test this please

**Pull request merged**